linopy/io.py

```diff
@@ -307,6 +307,93 @@ def to_file(m, fn, integer_label="general"):
     return fn
 
 
+def to_mosekpy(model, task=None):
+    """
+    Export model to MOSEK.
+
+    Export the model directly to MOSEK without writing files.
+
+    Parameters
+    ----------
+    m : linopy.Model
+    task : empty MOSEK task
+
+    Returns
+    -------
+    task : MOSEK Task object
+    """
+
+    import mosek
+
+    if task is None:
+        task = mosek.Task()
+
+    task.appendvars(model.nvars)
+    task.appendcons(model.ncons)
+
+    M = model.matrices
+    for j, n in enumerate(("x" + M.vlabels.astype(str).astype(object))):
+        task.putvarname(j, n)
+
+    ## Variables
+
+    bkx = [
+        (
+            (
+                (mosek.boundkey.ra if l < u else mosek.boundkey.fx)
+                if u < np.inf
+                else mosek.boundkey.lo
+            )
+            if (l > -np.inf)
+            else (mosek.boundkey.up if (u < np.inf) else mosek.boundkey.fr)
+        )
+        for (l, u) in zip(M.lb, M.ub)
+    ]
```

Comment:
Could you short say what is happening here? It's a bit hard to read for me.

Reply:
I have inserted a comment now. MOSEK used a (boundkey,lb,ub) triplet to define a bound to avoid using `inf`. Here I compute the bound keys based on the bound values.